### PR TITLE
My Site Dashboard: Phase 1 - Quick Actions

### PR DESCRIPTION
**Pull request by zwarm**
Parent #15193

This PR includes
- UI changes according to the latest zeplin design docs (margins/paddings, font size, card border, button border)
- Rename `quick_start_toolbar` to `my_site_card_toolbar`
- Added a new layout file for the quick actions called `my_site_quick_actions_card`. This is the wrapper and includes the existing `quick_actions_block` layout file and the `my_site_card_toolbar`. Perhaps we **should** revisit names?
- Extracted the building of QuickActionsBlock to it's own builder `QuickActionsBlockBuilder` + associated tests in `QuickActionsBlockBuilderTest`.
- Updated unit tests in `MySiteViewModelTest`

Dark | Light
-- | --
Alt desc | Alt desc

**To test:**

**Test Quick Actions Clicks are functional**
- Launch the app 
- Navigate to My Site -> App Settings -> Test feature configuration
- Enable the `MySiteImprovementsFeatureConfig` and tap the button to restart the app 
- Navigate to the My Site tab
- Tap on all the quick starts button to ensure they are working correctly

**Test that Pages doesn't show in quick actions**
- Navigate to My Site
- Using the site selector, select a site that you don't have edit capability for
- On the My Site tab, the quick actions card should not show the Pages action

**Test that Pages action shows blue focus icon**
- Navigate to My Site
- Using the site selector, select a site that Quick Tasks are enabled
- Tap on the Customize Your Site to launch the dialog
- Tap on Review Site Pages
- Note that the Pages quick action button shows the flashing blue focus point.
 
**Test that Pages action shows blue focus icon**
- Navigate to My Site
- Using the site selector, select a site that Quick Tasks are enabled
- Tap on the Grow your audience to launch the dialog
- Tap on Check your stats
- Note that the Stats quick action button shows the flashing blue focus point.

**Note**
Make toolbar "More" invisible - this part will be handled separately.

## Regression Notes
1. Potential unintended areas of impact
Quick Actions navigation does not work

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
`MySiteViewModelTest` and `QuickActionsBlockBuilderTest`

PR submission checklist:

- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by ashiagr — APPROVED** [review]
Great job @zwarm! 🤩 
As hiding more menu is a separate sub-task, we're good to merge this PR. Will wait for approval from Chris.

**Review by osullivanchris — CHANGES_REQUESTED** [review]
Really nice to see this, it looks great! Just one small tweak. The space below the Quick Actions looks a bit larger than the mock. 

Screenshot 2021-08-26 at 12 48 46


I'm measuring it at 35dp but I'm guessing its actually 32dp either in one chunk or maybe there's 24 defined + 8 coming from within the Quick Actions component.

Could we reduce the space by 8dp there?

**Comment by osullivanchris** [comment]
I have also internally requested a user facing name for this feature in the comments at p2JRYi-5sv-p2

**Comment by ashiagr** [comment]
> Could we reduce the space by `8dp` there?

Reduced in a4eeab2b38cc54bc2373c059d8545264419ddd56. Chris can you take another look?

Before | After
-------|------
device-2021-08-26-173246 | device-2021-08-26-173205 



> I have also internally requested a user facing name for this feature

👍

**Review by osullivanchris — APPROVED** [review]
> Reduced in a4eeab2. Chris can you take another look?

Thanks @ashiagr. Seeing that it was 40->32 I actually think we can shave a little further and remove another 8dp making it 24dp if thats ok? Sorry to get you to make the same fix again. I think then it will be perfectly balanced.

I also noticed one other thing. When one of the actions is pressed, it displays a shadow like we had before. Could it instead have the grey ripple/overlay touch effect that we use on rows and most places?

Screenshot_2021-08-26-14-36-06-245_org wordpress android prealpha


I am hitting approve on this one because the details are minor, so you can merge before waiting for me to review again.

**Comment by ashiagr** [comment]
> Seeing that it was 40->32 I actually think we can shave a little further and remove another 8dp making it 24dp

You're right, it feels perfect by reducing by another 8dp. Reduced in 5def07f.

> I also noticed one other thing. When one of the actions is pressed, it displays a shadow like we had before. Could it instead have the grey ripple/overlay touch effect that we use on rows and most places?

Added in fa89508. It looks better indeed 🙌.

https://user-images.githubusercontent.com/1405144/131074733-d87ba8f8-f76a-45fd-950c-53fdcb8f1fa6.mov

**Comment by osullivanchris** [comment]
> You're right, it feels perfect by reducing by another 8dp. Reduced in 5def07f.

🙌  great, glad you agree too!

> Added in fa89508. It looks better indeed 🙌.

It definitely looks better. But I'm now seeing the grey I expected *and* the shadow is still there. Is it possible to remove the shadow? I believe these buttons were originally derived from Floating Action Buttons IIRC, so that could be causing the trouble 😄

**Comment by ashiagr** [comment]
> I expected and the shadow is still there

I see, hope commit 9403a6f resolves it 😄🤞 

https://user-images.githubusercontent.com/1405144/131098873-f2ff8c69-e747-4c76-9ce5-d5fe984e1ff7.mov

**Comment by osullivanchris** [comment]
> I see, hope commit 9403a6f resolves it 😄🤞

That's perfect. Thanks for resolving these little details @ashiagr !

**Comment by zwarm** [comment]
Thanks @ashiagr for addressing @osullivanchris feedback while I was AFK. 👍

File: WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.kt
Comment (PACE210) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.kt:435]: False
Stop

Comment (PACE210) [reply WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.kt:435]: False
Stop

**Pull request merged**
